contracts/main/stablecoin-core/StableSwapModuleWrapper.sol

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+
+import "../interfaces/IToken.sol";
+import "../interfaces/IStablecoinAdapter.sol";
+import "../interfaces/IStablecoin.sol";
+import "../interfaces/IBookKeeper.sol";
+import "../interfaces/IStableSwapModule.sol";
+import "../utils/SafeToken.sol";
+import "../interfaces/IStableSwapModuleWrapper.sol";
+ 
+contract StableSwapModuleWrapper is PausableUpgradeable, ReentrancyGuardUpgradeable, IStableSwapModuleWrapper{
+    using SafeToken for address;
+    uint256 internal constant WAD = 10 ** 18;
+
+    IBookKeeper public bookKeeper;
+
+    address public stablecoin;
+    address public token;
+    address public stableSwapModule;
+    bool public isDecentralizedState;
+    uint256 public totalValueDeposited;
+    
+
+    mapping(address => uint256) public depositTracker;
+    mapping(address => bool) public usersWhitelist;
+    
+    event LogDepositTokens(address indexed _depositor, uint256 _amount);
+    event LogWithdrawTokens(address indexed _depositor, uint256 _amount);
+    event LogAddToWhitelist(address indexed user);
```

Comment:
**_Please use same naming convention for events, may create small issues on frontend and in testing_**